fedimint-bitcoind/src/bitcoincore_rpc.rs

```diff
@@ -292,3 +299,112 @@ impl IBitcoindRpc for ElectrumClient {
+        BitcoinRpcBackendType::Esplora
+    }
+
+    async fn get_network(&self) -> Result<Network> {
+        let genesis_height: u32 = 0;
+        let genesis_hash = self.0.get_block_hash(genesis_height).await?;
+
+        // TODO: How could we extract the BlockHash to a constant, and use something like the `bitcoin::blockdata::constants::genesis_block` from rust-bitcoin ?
+        let network = match genesis_hash.to_hex().as_str() {
+            "000000000019d6689c085ae165831e934ff763ae46a2a6c172b3f1b60a8ce26f" => Network::Bitcoin,
+            "000000000933ea01ad0ee984209779baaec3ced90fa3f408719526f8d77f4943" => Network::Testnet,
+            "00000008819873e925422c1ff0f99f7cc9bbb232af63a077a480a3633bee1ef6" => Network::Signet,
+            hash => {
+                warn!("Unknown genesis hash {hash} - assuming regtest");
+                Network::Regtest
+            }
+        };
+
+        Ok(network)
+    }
+
+    async fn get_block_height(&self) -> Result<u64> {
+        match self.0.get_height().await {
+            Ok(height) => Ok(height as u64),
+            Err(e) => Err(e.into()),
+        }
+    }
+
+    async fn get_block_hash(&self, height: u64) -> Result<BlockHash> {
+        self.0
+            .get_block_hash(height as u32)
+            .await
+            .map_err(anyhow::Error::from)
+            .map_err(Into::into)
+    }
+
+    async fn get_block(&self, hash: &BlockHash) -> Result<Block> {
+        self.0
+            .get_block_by_hash(hash)
+            .await
+            .map(|block| {
+                block.ok_or_else(|| {
+                    anyhow::Error::msg(format!(
+                        "The fetched block for given height {hash} has not been not found"
+                    ))
+                })
+            })?
+            .map_err(anyhow::Error::from)
+            .map_err(Into::into)
+    }
+
+    async fn get_fee_rate(&self, confirmation_target: u16) -> Result<Option<Feerate>> {
+        let fee_estimates: HashMap<String, f64> = self.0.get_fee_estimates().await?;
+
+        let fee_rate_vb =
+            esplora_client::convert_fee_rate(confirmation_target.into(), fee_estimates)?;
+
+        let fee_rate_kvb = fee_rate_vb * 1_000f32;
+
+        Ok(Some(Feerate {
+            sats_per_kvb: (fee_rate_kvb).ceil() as u64,
+        }))
+    }
+
+    async fn submit_transaction(&self, transaction: Transaction) -> Result<()> {
+        self.0
+            .broadcast(&transaction)
+            .await
+            .map_err(anyhow::Error::from)
+            .map_err(Into::into)
+    }
+
+    async fn was_transaction_confirmed_in(
+        &self,
+        transaction: &Transaction,
+        height: u64,
+    ) -> Result<bool> {
+        if self.get_block_height().await? <= height {
+            bail!("Esplora backend does not contain the block at {height} yet");
+        }
+
+        let txid = transaction.txid();
+        match self.0.get_tx_status(&txid).await? {
```

Comment:
This has different semantics then other implementations. This will respond `true` even if `transaction` was confirmed in a different block (even later one).

`was_transaction_confirmed_in` was a workaround for lack of `get_block` in Electrum. You don't have to implement it if `get_block` is available

Reply:
You could actually filter by block height here, but I agree it's safer to make this `unimplemented`.

Reply:
Ah, totally missed the implementation semantics 😅. Sure, I'll use the `get_block` instead and keep `was_transaction_confirmed_in` unimplemented.